### ISSUE TITLE
mapbox-gl instance 'on' event 'load', loaded() member not work when upgrade to version 2.11.0, but they work well in version 2.10.0

### ISSUE THREAD

**Issue by Jackclarify** [jsreport/jsreport#813]
Environment:
window10 pro, Version 10.0.19043 Build 19043
node 12.6.0, npm 6.13.4, jsreport 2.11.0

Here is my script in my handlebar below:
```
<script>
    console.log('aaaaaaaaaaaaaaaaaaaaaaa');
    var mapboxglConfig = {
        accessToken: 'xxx',
        container: '{{Report.index}}',
        style: 'mapbox://styles/mapbox/light-v10',
        zoom: 9,
        bounds: {{Report.index}}.fitBounds,
        fitBoundsOptions: { padding: { top: 70, bottom: 70, left: 20, right: 20 } },
        fadeDuration: 0,
        interactive: false,
        preserveDrawingBuffer: true
    }
    var map_{{Report.index}} = new mapboxgl.Map(mapboxglConfig);

    map_{{Report.index}}.on('load', function() {
        console.log('bbbbbbbbbbbbbbbbbb');
    });

    console.log('______________map_{{spotlightReport.index}}.on');
    setInterval(function() {
        console.log(map_{{Report.index}}.loaded())
    }, 300);
    console.log('_________aaaaaaa_____map_{{spotlightReport.index}}.on');

    map_{{Report.index}}.on('render', function() {
        console.log('ccccccccccccccccccccccc');
    });

</script>
```
but version 2.10.0 works well, could some body give me some help on this issue? Thanks!

**Comment by pofider** [comment]
The 2.11.0 didn't include a change regarding the chrome version or how jsreport operates with it.

Could you prepare a simple playground demo replicating the issue? Thank you
https://playground.jsreport.net/

**Comment by Jackclarify** [comment]
@pofider Okay, I will try. After further investigate, I find it might relate to package jsreport-chrome-pdf. It works well with version 1.9.0, and does not work with version 1.10.0.

Thank you for your quickly reply. Thanks!

**Comment by pofider** [comment]
I guess you mean version 2.10.0, not 1.10.0.

The change was introduced in 2.10.0 prevents access to local files unless `allowLocalFilesAccess` is set to true in the root config.

**Comment by Jackclarify** [comment]
Yes, you're right! release: 2.10.0 tag: 1.10.0 in github. When I runing npm i jsreport-chrome-pdf@2.9.0, it throw error:
npm ERR! code ETARGET
npm ERR! notarget No matching version found for jsreport-chrome-pdf@2.9.0.

My jsreport configuration is below:

`{
    'rootDirectory': rootDirectory,
    'tempDirectory': path.join(rootDirectory, 'reportserver/data/temp'),
    'loadConfig': false,
    'store': {
        'provider': 'fs'
    },
    'certificate': {
        'key': 'certificates/jsreport.net.key',
        'cert': 'certificates/jsreport.net.cert'
    },
    'blobStorage': {
        'provider': 'fs'
    },
    'httpPort': 5488,
    'httpsPort': null,
    'allowLocalFilesAccess': true,
    "reportTimeout": 450000,
    'templatingEngines': {
        'strategy': 'in-process',
        'numberOfWorkers': 1,
        'allowedModules': []
    },
    'logger': {
        'console': {
            'transport': 'console',
            'level': 'debug'   // debug
        },
        'file': {
            'transport': 'file',
            'level': 'info',
            'filename': 'logs/reporter.log'
        },
        'error': {
            'transport': 'file',
            'level': 'error',
            'filename': 'logs/error.log'
        }
    },
    'extensions': {
        'assets': {
            'allowedFiles': 'assets/content/**/*',
            'searchOnDiskIfNotFoundInStore': true,
            'rootUrlForLinks': 'file:///' + rootDirectory + 'reportserver/data/'
        },
        'fs-store': {
            'dataDirectory': path.join(rootDirectory, 'reportserver/data'),
            'compactionInterval': 60000
        },
        'chrome-pdf': {
            'launchOptions': {
                'defaultViewport': null,
                'args': ['--no-sandbox', '--window-size=1800,1350']
            },
            'strategy': 'chrome-pool',
            'allowLocalFilesAccess': true,
            'numberOfWorkers': 3
        },
        'express': {
            'enabled': false
        },
        'scripts': {
            'allowedModules': [
                'request',
                'moment',
                'lodash',
                'util',
                './reportserver/lib/Injector',
                './reportserver/build/main/index',
                './reportserver/lib/ngDataProvider',
                './reportserver/build/main/shared/commonts/index'
            ],
            'numberOfWorkers': 1,
            'timeout': 450000
        },
        'html-to-xlsx': {
            'enabled': false
        }
    }
}`

**Comment by Jackclarify** [comment]
@pofider Here is a small example: https://playground.jsreport.net/w/Jackclarify/kMAhgZAW
Could you please help to review it?
Any question please let me know.
Thank you!

**Comment by pofider** [comment]
Thank you for the demo.

I narrowed down the problem to the bug in the chrome driver https://github.com/puppeteer/puppeteer/issues/5003
It seems mapbox is using `blob:` protocol which hangs in conjunction with the driver request interceptors.
Unfortunately, we need to use the request interceptor (added in the latest jsreport-chrome-pdf). There is no solution for this now.

I recommend downgrading to the jsreport-chrome-pdf@1.9.0 until this puppeteer bug gets resolved.
I apologize for the inconvenience.

**Comment by Jackclarify** [comment]
Okay, thank you for your reply. I have downgraded to jsreport-chrome-pdf@1.9.0. 

And look forward to upgrading to the latest version of puppeteer. 

Because I encountered a problem, when using puppeteer@1.7.0 in jsreport@2.2.0 in the unit test, the pdfs are not equal in some case. It seems to be a coordinate deviation when comparing base64 or slices(using the package compare-pdf@1.1.4 to compare pdf). It works well after upgrading to puppeteer 10.0.0.

Thank you @pofider !
